app-modules/consent/graphql/user-consent-agreement.graphql

```diff
@@ -0,0 +1,118 @@
+type UserConsentAgreement
+    @model(class: "AdvisingApp\\Consent\\Models\\UserConsentAgreement") {
+    id: UUID!
+
+    ip_address: String!
+
+    created_at: DateTime
+
+    updated_at: DateTime
+
+    deleted_at: DateTime
+
+    user: User! @belongsTo
+
+    consent_agreement: ConsentAgreement!
+        @belongsTo(relation: "consentAgreement")
+
+    audits: [Audit!] @hasMany
+
+    audits_count: Int @count(relation: "audits")
+}
+
+input UserConsentAgreementsQuery {
+    id: UUID
+
+    ip_address: String
+
+    created_at: DateTime
+
+    updated_at: DateTime
+
+    deleted_at: DateTime
+
+    user: UsersQuery
+
+    consent_agreement: ConsentAgreementsQuery
+
+    audits: AuditsQuery
+}
+
+type UserConsentAgreementQueries {
+    "Get a specific user consent agreement by ID."
+    find(
+        id: UUID!
+            @whereKey
+            @rules(
+                apply: ["required", "uuid", "exists:user_consent_agreements"]
+            )
+    ): UserConsentAgreement
+        @find
+        @softDeletes
+        @canModel(
+            model: "AdvisingApp\\Consent\\Models\\ConsentAgreement"
+            ability: "viewAny"
+        )
+
+    "List multiple user consent agreements."
+    list(
+        "Filter by the user consent agreements attributes and relations."
+        where: UserConsentAgreementsQuery @searchBy
+        order: UserConsentAgreementsQuery @sortBy
+    ): [UserConsentAgreement!]!
+        @paginate
+        @softDeletes
+        @canModel(
+            model: "AdvisingApp\\Consent\\Models\\ConsentAgreement"
+            ability: "viewAny"
+        )
+}
+
+extend type Query {
+    userConsentAgreement: UserConsentAgreementQueries! @namespaced
+}
+
+input CreateUserConsentAgreementInput {
+    "The IP address of the user."
+    ip_address: String! @rules(apply: ["required", "string", "ip"])
```

Comment:
The frontend calls `$user->consentTo($consentAgreement);` inside of that method the request ip is set. GraphQL can't understand this by default because Lighthouse is trying to call the `update` method on the `ConsentAgreement`.

`UpdateUserConsentAgreementBelongsToManyInput` is used to enable the [nested mutation attachment functionality](https://lighthouse-php.com/6/eloquent/nested-mutations.html#hasmany) that Lighthouse enables. I don't know that I like this functionality because it looks a little weird, but it lets you consent multiple users at the same time. The order of operations is basically `consentAgreement->update(); consentAgreement->attach()' This does not go through `user->consentTo` so it doesn't understand how to set the request ip.

I've created a custom resolver that handles the update and attach action as follows:
```GraphQL
mutation ConsentAgreement2 {
    consentAgreement {
        update(
            id: "9b5a4ec0-cb85-4fde-be52-5f0feaa0fe09"
            input: {
                description: "123"
                users: { consent: ["9b5a4ec0-3512-4ce9-8a20-9e9de841852b"] }
            }
        ) {
            id
            type
            title
            description
            body
            created_at
            updated_at
            deleted_at
            user_consent_agreements_count
            user_consent_agreements {
                id
                ip_address
                created_at
                updated_at
                deleted_at
                audits_count
            }
        }
    }
}
```
I believe this is important to allow the nested mutations, just not necessarily in the way Lighthouse automatically provides.